src/Modal.js

```diff
@@ -269,15 +272,21 @@ class ModalManager {
         }
 
         this._reRender();
-        return {close: closeDialog};
+        return {
+            close: closeDialog,
+            then: (resolve, reject) => onFinishedProm.then(resolve, reject),
```

Comment:
also a `catch` for encouraging good code practice? Likewise below

Reply:
This is just implementing the Thenables interface for async/await which doesn't care for catch: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await#Examples

but fair enough

Reply:
true, but in practice we try and not use `.then(null, (e) => console.error(e))` or whatever

Reply:
Actually `catch` just doesn't line up with Thenables, unless I'm missing something

Reply:
could make it so that the raw promise gets returned, the only diff at the user-facing level would be
`const [...] = await modal.onFinished`
but then it'd be a real promise you could call then/catch on if you weren't using async/await

Reply:
I guess just skip it for now. We can add it later if we actually need it.

Reply:
going to merge as is, minor detail to change it in future

Reply:
> true, but in practice we try and not use .then(null, (e) => console.error(e)) or whatever

wouldn't work here for Thenables, it'd have to be https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/resolve
`Promise.resolve(modal).catch(...)`